pygfunction/pipes.py

```diff
@@ -3613,3 +3616,209 @@ def _Nusselt_number_turbulent_flow(Re, Pr, fDarcy):
+        R_p_out = conduction_thermal_resistance_circular_pipe(
+            r_out_in, r_out_out, k_p_out)
+        # Fluid-to-fluid thermal resistance [m.K/W]
+
+        # Outer pipe
+        h_f_a_in, h_f_a_out = \
+            convective_heat_transfer_coefficient_concentric_annulus(
+                m_flow_pipe, r_in_out, r_out_in, fluid.mu, fluid.rho, fluid.k,
+                fluid.cp, epsilon)
+
+        # Coaxial GHE in borehole
+        R_f_out_out = 1.0 / (h_f_a_out * 2 * np.pi * r_out_in)
+
+        return R_p_out + R_f_out_out
+
+    else:
+        raise ValueError(f"Unsupported pipe_type: '{pipe_type.name}'")
+
+
+def compute_R_ff(pipe_type: PipeType, m_flow_borehole: float, r_in: Union[float, tuple, list],
+                 r_out: Union[float, tuple, list], k_p: Union[float, tuple, list], epsilon: float,
+                 fluid: Fluid) -> float:
+    if pipe_type == PipeType.COAXIAL_ANNULAR_IN:
+
+        m_flow_pipe = m_flow_borehole
+
+        r_in_in = r_in[1]
+        r_in_out = r_out[1]
+        r_out_in = r_in[0]
+        k_p_in = k_p[1]
+
+        # Inner pipe
+        R_p_in = conduction_thermal_resistance_circular_pipe(
+            r_in_in, r_in_out, k_p_in)
+
+        # Fluid-to-fluid thermal resistance [m.K/W]
+        # Inner pipe
+        h_f_in = convective_heat_transfer_coefficient_circular_pipe(
+            m_flow_pipe, r_in_in, fluid.mu, fluid.rho, fluid.k, fluid.cp, epsilon)
+        R_f_in = 1.0 / (h_f_in * 2 * np.pi * r_in_in)
+
+        # Outer pipe
+        h_f_a_in, h_f_a_out = \
+            convective_heat_transfer_coefficient_concentric_annulus(
+                m_flow_borehole, r_in_out, r_out_in, fluid.mu, fluid.rho, fluid.k,
+                fluid.cp, epsilon)
+        R_f_out_in = 1.0 / (h_f_a_in * 2 * np.pi * r_in_out)
+
+        return R_f_in + R_p_in + R_f_out_in
+
+    elif pipe_type == PipeType.COAXIAL_ANNULAR_OUT:
+
+        m_flow_pipe = m_flow_borehole
+
+        r_in_in = r_in[0]
+        r_in_out = r_out[0]
+        r_out_in = r_in[1]
+        k_p_in = k_p[1]
+
+        # Pipe thermal resistances [m.K/W]
+        # Inner pipe
+        R_p_in = conduction_thermal_resistance_circular_pipe(
+            r_in_in, r_in_out, k_p_in)
+
+        # Fluid-to-fluid thermal resistance [m.K/W]
+        # Inner pipe
+        h_f_in = convective_heat_transfer_coefficient_circular_pipe(
+            m_flow_pipe, r_in_in, fluid.mu, fluid.rho, fluid.k, fluid.cp, epsilon)
+        R_f_in = 1.0 / (h_f_in * 2 * np.pi * r_in_in)
+
+        # Outer pipe
+        h_f_a_in, h_f_a_out = \
+            convective_heat_transfer_coefficient_concentric_annulus(
+                m_flow_pipe, r_in_out, r_out_in, fluid.mu, fluid.rho, fluid.k,
+                fluid.cp, epsilon)
+        R_f_out_in = 1.0 / (h_f_a_in * 2 * np.pi * r_in_out)
+
+        return R_f_in + R_p_in + R_f_out_in
+
+    else:
+        raise ValueError(f"Unsupported pipe_type: '{pipe_type.name}'")
+
+
+def get_pipes(
+        boreholes: list[Borehole],
```

Comment:
I would suggest allowing Union[List[Borehole], Borefield] and adding at the top of the function:

```
if isinstance(boreholes, Borefield):
    boreholes = boreholes.to_boreholes()
```